server/src/test/java/org/kaazing/gateway/server/topic/DefaultClusterTopicTest.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.server.topic;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.kaazing.gateway.server.context.resolve.DefaultClusterContext;
+import org.kaazing.gateway.service.cluster.MemberId;
+import org.kaazing.test.util.ITUtil;
+
+import com.hazelcast.core.ITopic;
+import com.hazelcast.core.MessageListener;
+
+public class DefaultClusterTopicTest {
```

Comment:
I would expect that this test and StandaloneClusterTopicTest would be sharing the same code, probably by both extending the same abstract class,  because the behavior for topics should be the same for the two cases. The superclass can have a test rule or before and after methods to create and dispose the cluster contexts by calling abstract methods which would be implemented differently in the subclasses.